rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/project/Sw360ProjectService.java

```diff
@@ -1538,4 +1538,92 @@ public List<ReleaseLink> serveLinkedReleasesInDependencyNetworkByIndexPath(Strin
         ProjectService.Iface sw360ProjectClient = getThriftProjectClient();
         return sw360ProjectClient.getReleaseLinksOfProjectNetWorkByIndexPath(projectId, indexPath, sw360User);
     }
+
+    public JSONObject addLicenseToLinkedReleases(String projectId, User sw360User) throws TTransportException, TException {
+        ThriftClients thriftClients = new ThriftClients();
+        ProjectService.Iface projectClient = getThriftProjectClient();
+        LicenseInfoService.Iface licenseInfoClient = thriftClients.makeLicenseInfoClient();
+        ComponentService.Iface componentClient = thriftClients.makeComponentClient();
+        JSONObject jsonResult = new JSONObject();
+
+        try {
+            Project project = projectClient.getProjectById(projectId, sw360User);
+            if (project == null) {
+                throw new IllegalArgumentException("Project with ID " + projectId + " not found.");
+            }
+
+            Set<String> releaseIds = CommonUtils.getNullToEmptyKeyset(project.getReleaseIdToUsage());
+            List<Release> releasesWithSingleCLI = new ArrayList<>();
+            List<Release> releasesWithMultipleCLI = new ArrayList<>();
+
+            for (String releaseId : releaseIds) {
+                Release release = componentClient.getReleaseById(releaseId, sw360User);
+                if (release == null) {
+                    throw new IllegalArgumentException("Release with ID " + releaseId + " not found.");
+                }
+
+                List<Attachment> approvedCliAttachments = SW360Utils.getApprovedClxAttachmentForRelease(release);
+                if (approvedCliAttachments.isEmpty()) {
+                    approvedCliAttachments = SW360Utils.getClxAttachmentForRelease(release);
+                }
+
+                if (approvedCliAttachments.size() == 1) {
+                    processSingleAttachment(approvedCliAttachments.get(0), release, licenseInfoClient, sw360User);
+                    releasesWithSingleCLI.add(release);
+                } else {
+                    if (approvedCliAttachments.size() > 1) {
+                        releasesWithMultipleCLI.add(release);
+                    }
+                    jsonResult.put(SW360Constants.STATUS, SW360Constants.FAILURE);
+                }
+
+                componentClient.updateRelease(release, sw360User);
```

Comment:
This call can result in creation of Moderation Request. Please add comment field for the same in the controller as done in #2405

Reply:
@GMishx , For this endpoint, a Moderation Request is not required, so adding a comment field is not necessary.

Reply:
@nikkuma7 , I get it now, the function `processSingleAttachment()` updates the release by calling `setMainLicenseIds()` and `setOtherLicenseIds()`.

Thus, calling `componentClient.updateRelease()` can result in moderation request. The comparator for `Release` does checks the `mainLicenseIds` and `otherLicenseIds` fields.

Please test again!

Reply:
@nikkuma7 any updates on this?

Reply:
A moderation request is now created when clicking the "Add License Info" button. The moderation request is also created at the time of attachment creation or when modifying the attachment file. However, the moderation request is not created when clicking the "License Info" button.

Reply:
(from internal discussions) MR will not be created as this endpoint can be access only by Clearing Admin and above roles. Thus, this comment is not applicable.